packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -461,6 +492,26 @@ class SampleTypeDesignerImpl extends React.PureComponent<Props & InjectedBaseDom
         }
     };
 
+    onAddUniqueIdField = (fieldConfig: Partial<IDomainField>): void => {
+        this.setState((state) => ({
+            model: state.model.set("domain", addDomainField(this.state.model.domain, fieldConfig)) as SampleTypeModel,
+        }));
+    }
+
+    uniqueIdBannerRenderer = (config: IAppDomainHeader): ReactNode  => {
+        if (isCommunityDistribution()) {
+            return null;
+        }
+        return (
+            <UniqueIdBanner model={this.state.model} isFieldsPanel={true} onAddField={config.onAddField}/>
+        )
+    }
+
+    getNumNewUniqueIdFields(): number {
+        const { model } = this.state;
+        return model.domain.fields.filter(field => field.isNew() &&  field.isUniqueIdField()).count();
```

Comment:
I think this might be where we need to add the check for if this is a new domain / sample type vs an existing one that we are adding to.